src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java

```diff
@@ -237,14 +245,25 @@ private BigQuery getBigQuery() throws IOException {
 
 
   @Override
-  public ConnectorSpec generateSpec(ConnectorSpecRequest connectorSpecRequest) {
+  public ConnectorSpec generateSpec(ConnectorSpecRequest connectorSpecRequest) throws IOException {
     BigQueryPath path = new BigQueryPath(connectorSpecRequest.getPath());
-    String dataset = path.getDataset();
-    String table = path.getTable();
-    if (table == null) {
+    String datasetName = path.getDataset();
+    String tableName = path.getTable();
+    if (tableName == null) {
       throw new IllegalArgumentException("Path should contain both dataset and table name.");
     }
-    return ConnectorSpec.builder().addProperty(BigQuerySourceConfig.NAME_DATASET, dataset)
-      .addProperty(BigQuerySourceConfig.NAME_TABLE, table).build();
+
+    Map<String, String> properties = new HashMap<>();
```

Comment:
Can you also add the properties of useConnection flag and connection here? This is to ensure UI have all the properties needed when going to the source, you can find the example here: https://github.com/cdapio/hydrator-plugins/compare/file-connector#diff-52ec8fe889fbb4fe210de39c3f35ffbe41f27ba518b80a9d0a60f4d43906d554R139-R140

Reply:
done :)